### ISSUE TITLE
Reenabling unit tests for Pangea2 target

### ISSUE THREAD

**Issue by castelletto1** [GEOSX/GEOSX#1245]
Unit test `testLAOperations` fails on Pangea2 target. See discussion https://github.com/GEOSX/GEOSX/pull/1239#issue-524482520.

**Comment by TotoGaz** [comment]
Before it's deleted, here is the log. Only `testLAOperations` is failling.
```
         Start  78: testLAOperations
78: Test command: /data_local/sw/OpenMPI/RHEL7/2.1.5/gcc/8.3.0/bin/mpiexec "-n" "2" "--allow-run-as-root" "/tmp/build/tests/testLAOperations" "-x" "2"
78: Test timeout computed to be: 1500
78: [==========] Running 16 tests from 6 test suites.
78: [----------] Global test environment set-up.
78: [----------] 3 tests from Trilinos/LAOperationsTest/0, where TypeParam = geosx::TrilinosInterface
78: [ RUN      ] Trilinos/LAOperationsTest/0.VectorFunctions
78: [==========] Running 16 tests from 6 test suites.
78: [----------] Global test environment set-up.
78: [----------] 3 tests from Trilinos/LAOperationsTest/0, where TypeParam = geosx::TrilinosInterface
78: [ RUN      ] Trilinos/LAOperationsTest/0.VectorFunctions
78: [       OK ] Trilinos/LAOperationsTest/0.VectorFunctions (98 ms)
78: [ RUN      ] Trilinos/LAOperationsTest/0.MatrixMatrixOperations
78: [       OK ] Trilinos/LAOperationsTest/0.VectorFunctions (96 ms)
78: [ RUN      ] Trilinos/LAOperationsTest/0.MatrixMatrixOperations
78: [6cd5ab4340b7:06459] Read -1, expected 14400, errno = 1
78: [6cd5ab4340b7:06460] Read -1, expected 14400, errno = 1
78: [       OK ] Trilinos/LAOperationsTest/0.MatrixMatrixOperations (30 ms)
78: [ RUN      ] Trilinos/LAOperationsTest/0.RectangularMatrixOperations
78: [       OK ] Trilinos/LAOperationsTest/0.MatrixMatrixOperations (30 ms)
78: [ RUN      ] Trilinos/LAOperationsTest/0.RectangularMatrixOperations
78: [       OK ] Trilinos/LAOperationsTest/0.RectangularMatrixOperations (1 ms)
78: [----------] 3 tests from Trilinos/LAOperationsTest/0 (130 ms total)
78: 
78: [----------] 3 tests from Hypre/LAOperationsTest/0, where TypeParam = geosx::HypreInterface
78: [ RUN      ] Hypre/LAOperationsTest/0.VectorFunctions
78: [       OK ] Trilinos/LAOperationsTest/0.RectangularMatrixOperations (1 ms)
78: [----------] 3 tests from Trilinos/LAOperationsTest/0 (129 ms total)
78: 
78: [----------] 3 tests from Hypre/LAOperationsTest/0, where TypeParam = geosx::HypreInterface
78: [ RUN      ] Hypre/LAOperationsTest/0.VectorFunctions
78: [       OK ] Hypre/LAOperationsTest/0.VectorFunctions (3 ms)
78: [ RUN      ] Hypre/LAOperationsTest/0.MatrixMatrixOperations
78: [       OK ] Hypre/LAOperationsTest/0.VectorFunctions (2 ms)
78: [ RUN      ] Hypre/LAOperationsTest/0.MatrixMatrixOperations
78: [       OK ] Hypre/LAOperationsTest/0.MatrixMatrixOperations (10 ms)
78: [ RUN      ] Hypre/LAOperationsTest/0.RectangularMatrixOperations
78: [       OK ] Hypre/LAOperationsTest/0.MatrixMatrixOperations (9 ms)
78: [ RUN      ] Hypre/LAOperationsTest/0.RectangularMatrixOperations
78: [       OK ] Hypre/LAOperationsTest/0.RectangularMatrixOperations (1 ms)
78: [----------] 3 tests from Hypre/LAOperationsTest/0 (15 ms total)
78: 
78: [----------] 3 tests from Trilinos/SolverTestLaplace2D/0, where TypeParam = geosx::TrilinosInterface
78: [ RUN      ] Trilinos/SolverTestLaplace2D/0.Direct
78: [       OK ] Hypre/LAOperationsTest/0.RectangularMatrixOperations (1 ms)
78: [----------] 3 tests from Hypre/LAOperationsTest/0 (14 ms total)
78: 
78: [----------] 3 tests from Trilinos/SolverTestLaplace2D/0, where TypeParam = geosx::TrilinosInterface
78: [ RUN      ] Trilinos/SolverTestLaplace2D/0.Direct
78: [6cd5ab4340b7:06460] Read -1, expected 60000, errno = 1
78: [6cd5ab4340b7:06459] Read -1, expected 518384, errno = 1
78: [6cd5ab4340b7:06459] Read -1, expected 40000, errno = 1
78: [6cd5ab4340b7:06460] Read -1, expected 40000, errno = 1
78: [       OK ] Trilinos/SolverTestLaplace2D/0.Direct (669 ms)
78: [ RUN      ] Trilinos/SolverTestLaplace2D/0.GMRES_ILU
78: [       OK ] Trilinos/SolverTestLaplace2D/0.Direct (671 ms)
78: [ RUN      ] Trilinos/SolverTestLaplace2D/0.GMRES_ILU
78: [       OK ] Trilinos/SolverTestLaplace2D/0.GMRES_ILU (177 ms)
78: [ RUN      ] Trilinos/SolverTestLaplace2D/0.CG_AMG
78: [       OK ] Trilinos/SolverTestLaplace2D/0.GMRES_ILU (179 ms)
78: [ RUN      ] Trilinos/SolverTestLaplace2D/0.CG_AMG
78: [       OK ] Trilinos/SolverTestLaplace2D/0.CG_AMG (160 ms)
78: [----------] 3 tests from Trilinos/SolverTestLaplace2D/0 (1010 ms total)
78: 
78: [----------] 3 tests from Hypre/SolverTestLaplace2D/0, where TypeParam = geosx::HypreInterface
78: [ RUN      ] Hypre/SolverTestLaplace2D/0.Direct
78: [       OK ] Trilinos/SolverTestLaplace2D/0.CG_AMG (162 ms)
78: [----------] 3 tests from Trilinos/SolverTestLaplace2D/0 (1012 ms total)
78: 
78: [----------] 3 tests from Hypre/SolverTestLaplace2D/0, where TypeParam = geosx::HypreInterface
78: [ RUN      ] Hypre/SolverTestLaplace2D/0.Direct
78: [6cd5ab4340b7:06460] Read -1, expected 20000, errno = 1
78: [6cd5ab4340b7:06459] Read -1, expected 20000, errno = 1
78: [6cd5ab4340b7:06460] Read -1, expected 199192, errno = 1
78: [6cd5ab4340b7:06460] Read -1, expected 99596, errno = 1
78: [6cd5ab4340b7:06459] Read -1, expected 199192, errno = 1
78: [6cd5ab4340b7:06459] Read -1, expected 99596, errno = 1
78: [6cd5ab4340b7:06460] Read -1, expected 40000, errno = 1
78: [6cd5ab4340b7:06459] Read -1, expected 40000, errno = 1
78: Received signal 11: Segmentation fault
78: 
78: ** StackTrace of 16 frames **
78: Frame 0:  
78: Frame 1: hypre_VectorToParVector 
78: Frame 2: HYPRE_VectorToParVector 
78: Frame 3: geosx::SuiteSparseSolve(geosx::SuiteSparse&, geosx::HypreVector const&, geosx::HypreVector&, bool) 
78: Frame 4:  
78: Frame 5: geosx::HypreSolver::solve(geosx::HypreMatrix&, geosx::HypreVector&, geosx::HypreVector&, geosx::DofManager const*) 
78: Frame 6: SolverTestBase<geosx::HypreInterface>::test(geosx::LinearSolverParameters const&) 
78: Frame 7: gtest_suite_SolverTestLaplace2D_::Direct<geosx::HypreInterface>::TestBody() 
78: Frame 8: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) 
78: Frame 9: testing::Test::Run() 
78: Frame 10: testing::TestInfo::Run() 
78: Frame 11: testing::TestSuite::Run() 
78: Frame 12: testing::internal::UnitTestImpl::RunAllTests() 
78: Frame 13: testing::UnitTest::Run() 
78: Frame 14: main 
78: Frame 15: __libc_start_main 
78: Frame 16:  
78: =====
78: 
78: --------------------------------------------------------------------------
78: MPI_ABORT was invoked on rank 1 in communicator MPI_COMM_WORLD
78: with errorcode 1.
78: 
78: NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
78: You may or may not see output from other processes, depending on
78: exactly when Open MPI kills them.
78: --------------------------------------------------------------------------
78: Received signal 15: Terminated
78: 
78: ** StackTrace of 20 frames **
78: Frame 0:  
78: Frame 1: __sched_yield 
78: Frame 2: mca_pml_ob1_recv 
78: Frame 3: ompi_coll_base_gather_intra_basic_linear 
78: Frame 4: MPI_Gather 
78: Frame 5: hypre_VectorToParVector 
78: Frame 6: HYPRE_VectorToParVector 
78: Frame 7: geosx::SuiteSparseSolve(geosx::SuiteSparse&, geosx::HypreVector const&, geosx::HypreVector&, bool) 
78: Frame 8:  
78: Frame 9: geosx::HypreSolver::solve(geosx::HypreMatrix&, geosx::HypreVector&, geosx::HypreVector&, geosx::DofManager const*) 
78: Frame 10: SolverTestBase<geosx::HypreInterface>::test(geosx::LinearSolverParameters const&) 
78: Frame 11: gtest_suite_SolverTestLaplace2D_::Direct<geosx::HypreInterface>::TestBody() 
78: Frame 12: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) 
78: Frame 13: testing::Test::Run() 
78: Frame 14: testing::TestInfo::Run() 
78: Frame 15: testing::TestSuite::Run() 
78: Frame 16: testing::internal::UnitTestImpl::RunAllTests() 
78: Frame 17: testing::UnitTest::Run() 
78: Frame 18: main 
78: Frame 19: __libc_start_main 
78: Frame 20:  
78: =====
78: 
78: [6cd5ab4340b7:06454] 1 more process has sent help message help-mpi-api.txt / mpi-abort
78: [6cd5ab4340b7:06454] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
```

**Comment by TotoGaz** [comment]
Hi @castelletto1 when I try to compile in the Pangea2 docker I get this (some `HYPRE_BigInt` being an `int` and not a `long long int`, I think I've heard of that) :
```
FAILED: coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreMatrix.cpp.o 
/data_local/sw/gcc/RHEL7/8.3.0/bin/g++   -I../coreComponents/linearAlgebra -I../coreComponents -Iinclude -isystem /data_local/sw/OpenMPI/RHEL7/2.1.5/gcc/8.3.0/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/suitesparse/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/superlu_dist/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/trilinos/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/hypre/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/parmetis/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/metis/include -isystem /data_local/sw/intel/RHEL7/compilers_and_libraries_2019.3.199/linux/mkl/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/pugixml/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/conduit/include/conduit -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/caliper/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/chai/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/raja/include -Wall -Wextra   -Werror  -Wall -Wextra -Wpedantic -pedantic-errors -Wshadow -Wfloat-equal -Wcast-align -Wcast-qual   -fdiagnostics-color=always  -g -fPIC   -pthread -std=c++14 -MD -MT coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreMatrix.cpp.o -MF coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreMatrix.cpp.o.d -o coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreMatrix.cpp.o -c ../coreComponents/linearAlgebra/interfaces/hypre/HypreMatrix.cpp
../coreComponents/linearAlgebra/interfaces/hypre/HypreMatrix.cpp: In member function 'virtual void geosx::HypreMatrix::print(std::ostream&) const':
../coreComponents/linearAlgebra/interfaces/hypre/HypreMatrix.cpp:1237:20: error: format '%lli' expects argument of type 'long long int', but argument 5 has type 'HYPRE_BigInt' {aka 'int'} [-Werror=format=]
                    "%11i%20lli%20lli%24.10e\n",
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../coreComponents/linearAlgebra/interfaces/hypre/HypreMatrix.cpp:1240:20:
                    col_map_offdiag[ offdiag_JA[j] ],
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[261/534] Building CXX object coreComp...dir/interfaces/hypre/HypreVector.cpp.o
FAILED: coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreVector.cpp.o 
/data_local/sw/gcc/RHEL7/8.3.0/bin/g++   -I../coreComponents/linearAlgebra -I../coreComponents -Iinclude -isystem /data_local/sw/OpenMPI/RHEL7/2.1.5/gcc/8.3.0/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/suitesparse/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/superlu_dist/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/trilinos/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/hypre/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/parmetis/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/metis/include -isystem /data_local/sw/intel/RHEL7/compilers_and_libraries_2019.3.199/linux/mkl/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/pugixml/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/conduit/include/conduit -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/caliper/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/chai/include -isystem /workrd/SCR/GEOSX/install/gcc8/GEOSX_TPL-c57d888/raja/include -Wall -Wextra   -Werror  -Wall -Wextra -Wpedantic -pedantic-errors -Wshadow -Wfloat-equal -Wcast-align -Wcast-qual   -fdiagnostics-color=always  -g -fPIC   -pthread -std=c++14 -MD -MT coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreVector.cpp.o -MF coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreVector.cpp.o.d -o coreComponents/linearAlgebra/CMakeFiles/linearAlgebra.dir/interfaces/hypre/HypreVector.cpp.o -c ../coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp
../coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp:35:39: error: static assertion failed: HYPRE_BigInt and geosx::globalIndex must have the same size
 static_assert( sizeof( HYPRE_BigInt ) == sizeof( globalIndex ),
                ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../coreComponents/linearAlgebra/interfaces/VectorBase.hpp:22,
                 from ../coreComponents/linearAlgebra/interfaces/hypre/HypreVector.hpp:22,
                 from ../coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp:19:
../coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp: In member function 'virtual void geosx::HypreVector::set(geosx::globalIndex, geosx::real64)':
../coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp:194:67: error: cannot convert 'const globalIndex*' {aka 'const long long int*'} to 'const HYPRE_BigInt*' {aka 'const int*'}
   GEOSX_LAI_CHECK_ERROR( HYPRE_IJVectorSetValues( m_ij_vector, 1, &globalRow, &value ) );
                                                                   ^~~~~~~~~~
```
This is strange that it was OK on travis.
I use the current version of the TPLs...
Do you have any idea why it's not working on my box?

**Comment by castelletto1** [comment]
@TotoGaz This is very strange. The integer issue was related to running hypre on GPU. It shouldn't have impacted the current version of the TPLs. @rrsettgast could you confirm?

**Comment by TotoGaz** [comment]
It's compiling with my standard `ubuntu` docker image. 🤷 

Inside of `hypre` there is this
```
#if defined(HYPRE_BIGINT)
typedef long long int HYPRE_BigInt;
typedef long long int HYPRE_Int;
#define HYPRE_MPI_BIG_INT MPI_LONG_LONG_INT
#define HYPRE_MPI_INT MPI_LONG_LONG_INT

#elif defined(HYPRE_MIXEDINT)
typedef long long int HYPRE_BigInt;
typedef int HYPRE_Int;
#define HYPRE_MPI_BIG_INT MPI_LONG_LONG_INT
#define HYPRE_MPI_INT MPI_INT

#else /* default */
typedef int HYPRE_BigInt;
typedef int HYPRE_Int;
#define HYPRE_MPI_BIG_INT MPI_INT
#define HYPRE_MPI_INT MPI_INT
#endif
```
Do I have to define something?

And what about this in `HypreVector.cpp`
```c++
static_assert( sizeof( HYPRE_BigInt ) == sizeof( globalIndex ),
               "HYPRE_BigInt and geosx::globalIndex must have the same size" );
```
How could this even compile without any action on `HYPRE_BigInt`? 🤔

**Comment by castelletto1** [comment]
> For my `Pangea2` image, it does not seem to be "out of the box".
> In the TPLs of Pangea2, I can read
> 
> ```
> /* Define to 1 if using long long int for HYPRE_BigInt */
> /* #undef HYPRE_MIXEDINT */
> ```
> 
> which is quite surprising sine the `--enable-mixedint` is provided when building `hypre`...
> 
> Isn't is strange?
> Are there specific actions to activate this option on GEOSX's side, where I could have a look?

Yes, this shouldn't happen. You should (in  `HYPRE_config.h`):

```c
 44 /* Define to 1 if using long long int for HYPRE_BigInt */
 45 #define HYPRE_MIXEDINT 1
 46 
 47 /* Define to 1 if using long long int for HYPRE_Int and HYPRE_BigInt */
 48 /* #undef HYPRE_BIGINT */
```

It seems that hypre has been built using the default option as to integers, i.e. `int` for both `HYPRE_Int` and `HYPRE_BigInt`.

**Comment by TotoGaz** [comment]
Indeed but the command line (from travis) looks good, and when I run it in my container the `HYDRE_config.h` is OK... 🤔 
Also, how could the build from `GEOSX` be successful itself?...

**Comment by TotoGaz** [comment]
Alright, I mixed up things with multiple containers. The current one has the proper `#define`.
But I cannot reproduce the error in my container...

**Comment by TotoGaz** [comment]
Found this https://github.com/open-mpi/ompi/issues/4948 about our 
```
79: [a071a014f61e:05550] Read -1, expected 10824, errno = 1
```
I do have a lot of this kind of messages (on `travis`, they are far fewer 🤷 ).
I did the suggested 
```shell
export OMPI_MCA_btl_vader_single_copy_mechanism=none
```
and I do not have this message again. Good. I shall try to understand what's going on and propose a fix for this.

Then in my test I can now see (which was already there but hidden by so much noise)
```
79: [----------] 2 tests from Trilinos/SolverTestElasticity2D/0, where TypeParam = geosx::TrilinosInterface
79: [ RUN      ] Trilinos/SolverTestElasticity2D/0.Direct
79: [       OK ] Trilinos/SolverTestElasticity2D/0.Direct (377 ms)
79: [ RUN      ] Trilinos/SolverTestElasticity2D/0.GMRES_AMG
79: [       OK ] Trilinos/SolverTestElasticity2D/0.Direct (377 ms)
79: [ RUN      ] Trilinos/SolverTestElasticity2D/0.GMRES_AMG
79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
79: [       OK ] Trilinos/SolverTestElasticity2D/0.GMRES_AMG (121 ms)
79: [----------] 2 tests from Trilinos/SolverTestElasticity2D/0 (498 ms total)
```
I know it's for `Trilinos` only (not `hypre`) and it's during `GMRES_AMG`, not `Direct`, but
- It says error and the test passes?
- Could some size error be an explanation for segfault in different contexts?

**Comment by castelletto1** [comment]
> Found this [open-mpi/ompi#4948](https://github.com/open-mpi/ompi/issues/4948) about our
> 
> ```
> 79: [a071a014f61e:05550] Read -1, expected 10824, errno = 1
> ```
> 
> I do have a lot of this kind of messages (on `travis`, they are far fewer 🤷 ).
> I did the suggested
> 
> ```shell
> export OMPI_MCA_btl_vader_single_copy_mechanism=none
> ```
> 
> and I do not have this message again. Good. I shall try to understand what's going on and propose a fix for this.

I think we had already done this: https (see discussion [here](https://github.com/GEOSX/GEOSX/pull/1239#issue-524482520)):

https://github.com/GEOSX/GEOSX/blob/43fc91e51e5d9be51457d4dee49e13c613f025db/.travis.yml#L47

**Comment by castelletto1** [comment]
> Then in my test I can now see (which was already there but hidden by so much noise)
> 
> ```
> 79: [----------] 2 tests from Trilinos/SolverTestElasticity2D/0, where TypeParam = geosx::TrilinosInterface
> 79: [ RUN      ] Trilinos/SolverTestElasticity2D/0.Direct
> 79: [       OK ] Trilinos/SolverTestElasticity2D/0.Direct (377 ms)
> 79: [ RUN      ] Trilinos/SolverTestElasticity2D/0.GMRES_AMG
> 79: [       OK ] Trilinos/SolverTestElasticity2D/0.Direct (377 ms)
> 79: [ RUN      ] Trilinos/SolverTestElasticity2D/0.GMRES_AMG
> 79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
> 79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
> 79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
> 79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
> 79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
> 79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
> 79: ERROR : in almalgamation 10200 => 5100(5100).  Have you specified the correct number of DOFs?
> 79: [       OK ] Trilinos/SolverTestElasticity2D/0.GMRES_AMG (121 ms)
> 79: [----------] 2 tests from Trilinos/SolverTestElasticity2D/0 (498 ms total)
> ```
> 
> I know it's for `Trilinos` only (not `hypre`) and it's during `GMRES_AMG`, not `Direct`, but
> 
> * It says error and the test passes?
> * Could some size error be an explanation for segfault in different contexts?

This is definitely worth investigating more.

**Comment by TotoGaz** [comment]
Oh yes, i did not copy it 🤦

**Comment by castelletto1** [comment]
Another element I should have mentioned (maybe supporting you size error concern). At some point I was able to have the Pangea2 target passing on Travis by adding for debugging purposes 

```c++
GEOSX_LOG_RANK_VAR( partitioning[0] );
GEOSX_LOG_RANK_VAR( partitioning[1] );
```

at line 162 below

https://github.com/GEOSX/GEOSX/blob/43fc91e51e5d9be51457d4dee49e13c613f025db/src/coreComponents/linearAlgebra/interfaces/hypre/HypreSuiteSparse.cpp#L157-L167

**Comment by TotoGaz** [comment]
Hi @castelletto1 
For my understanding, does it make sense to run in parallel the direct solver with parallel parameter set to `false`?
```c++
LinearSolverParameters params_Direct()
{
  LinearSolverParameters parameters;
  parameters.solverType = geosx::LinearSolverParameters::SolverType::direct;
  parameters.direct.parallel = 0;
  return parameters;
}
```

**Comment by castelletto1** [comment]
`parallel` is a parameter used to select the direct solver type:

- `0`: umfpack (serial direct solver). The system matrix is gathered in a single process, the linear is solved in serial, and then the solution vector is scattered across processes
- `1`: SuperLU_Dist (parallel direct solver)

Note that option `0` was actually the only one for the Trilinos LAI (via the KLU solver) before PR GEOSX/GEOSX/pull/1169 got merged.

For testing physics solver capabilities (see integrated tests), umpfack is preferable since it can be used with default options. This is not the case for SuperLU_Dist.

**Comment by TotoGaz** [comment]
Thx @castelletto1 

I gave a shot at `parameters.direct.parallel = 1` instead of `0` and (luckily) [the test passed](https://travis-ci.com/github/GEOSX/GEOSX/jobs/456180618) 🤷‍♂️ 

So there should something wrong _emphasized by_ `HypreSolver::solve_serialDirect` (which is run iff `parameters.direct.parallel` is `0`).

Do you have any idea about that?

**Comment by castelletto1** [comment]
Originally the unit test had `parameters.direct.parallel = 1` but the unit tests didn't pass on Lassen (see issue #1238). As a temporary fix (both direct solvers should be checked with unit tests), we switched to `parameters.direct.parallel = 1`. The result was that the Pangea2 build had problems.

**Comment by TotoGaz** [comment]
Live debugging from`travis-ci` it appears that the test (`ctest --extra-verbose -V -R testLAOperations`) fails randomly...
Actually fails more or less 25% of the time when ran multiple times in a row.
